test/unit/test_compatibility_v2_qvm.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+import pytest
+
+from pyquil import Program
+from pyquil.compatibility.v2.api import QVM
+from pyquil.api._errors import QVMError
+from pyquil.api._qvm import validate_noise_probabilities, validate_qubit_list, prepare_register_list
+from pyquil.api import QCSClientConfiguration
+from pyquil.gates import MEASURE, X
+from pyquil.quilbase import Declare, MemoryReference
+
+
+def test_qvm__default_client(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration)
+    p = Program(Declare("ro", "BIT"), X(0), MEASURE(0, MemoryReference("ro")))
+    qvm.load(p.wrap_in_numshots_loop(1000))
+    qvm.run()
+    qvm.wait()
+    bitstrings = qvm.read_memory(region_name="ro")
+    assert bitstrings.shape == (1000, 1)
+
+
+def test_qvm_run_pqer(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration, gate_noise=(0.01, 0.01, 0.01))
+    p = Program(Declare("ro", "BIT"), X(0), MEASURE(0, MemoryReference("ro")))
+    qvm.load(p.wrap_in_numshots_loop(1000))
+    qvm.run()
+    qvm.wait()
+    bitstrings = qvm.read_memory(region_name="ro")
+    assert bitstrings.shape == (1000, 1)
+    assert np.mean(bitstrings) > 0.8
+
+
+def test_qvm_run_just_program(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration, gate_noise=(0.01, 0.01, 0.01))
+    p = Program(Declare("ro", "BIT"), X(0), MEASURE(0, MemoryReference("ro")))
+    qvm.load(p.wrap_in_numshots_loop(1000))
+    qvm.run()
+    qvm.wait()
+    bitstrings = qvm.read_memory(region_name="ro")
+    assert bitstrings.shape == (1000, 1)
+    assert np.mean(bitstrings) > 0.8
+
+
+def test_qvm_run_only_pqer(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration, gate_noise=(0.01, 0.01, 0.01))
+    p = Program(Declare("ro", "BIT"), X(0), MEASURE(0, MemoryReference("ro")))
+
+    qvm.load(p.wrap_in_numshots_loop(1000))
+    qvm.run()
+    qvm.wait()
+    bitstrings = qvm.read_memory(region_name="ro")
+    assert bitstrings.shape == (1000, 1)
+    assert np.mean(bitstrings) > 0.8
+
+
+def test_qvm_run_region_declared_and_measured(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration)
+    p = Program(Declare("reg", "BIT"), X(0), MEASURE(0, MemoryReference("reg")))
+    qvm.load(p.wrap_in_numshots_loop(100)).run().wait()
+    bitstrings = qvm.read_memory(region_name="reg")
+    assert bitstrings.shape == (100, 1)
+
+
+def test_qvm_run_region_declared_not_measured(client_configuration: QCSClientConfiguration):
+    qvm = QVM(client_configuration=client_configuration)
+    p = Program(Declare("reg", "BIT"), X(0))
+    qvm.load(p.wrap_in_numshots_loop(100)).run().wait()
+    bitstrings = qvm.read_memory(region_name="reg")
+    assert bitstrings is None
+
+
+# For backwards compatibility, we support omitting the declaration for "ro" specifically
```

Comment:
This feature, and related tests, were removed recently before this PR. Is the intention to reinstate it for the sake of v2 compat? I'm OK either way, just wanted to make sure it was intentional and not just an artifact of starting this PR before that change was merged in.

Reply:
Depending on your answer, you may want to update the changelog at:
```
- The `ro` memory region is no longer implicitly declared. All memory regions must be declared explicitly.
```
to be explicit about whether this applies to the `compatibility` package.

Reply:
Some of that work in removing automatic `ro` declarations involved changes to `Program` (see [this](https://github.com/rigetti/pyquil/pull/1337/commits/26fcd792816bf343f21c3e4a23e5be303efac4ad)), and I don't see you vendoring a `compatibility` version of `Program`, so I'm unsure how this test passes. I haven't looked deeply though.

Reply:
Hang on... I just actually read the test assertion. All good. I think we just need to remove the confusing comment above the test.

Reply:
yeah you're right, doc fix here